lib/cdm_utils.rb

```diff
@@ -172,47 +172,57 @@ def self.conform(doc, collection_file_name, target_dir)
-      replace26 = replace25.gsub("<Personal_Name/>", "<Personal_Names/>")
-      replace27 = replace26.gsub("<Organization/>", "<Organization_Building/>")
-      replace28 = replace27.gsub("<Organization-Building/>", "<Organization_Building/>")
-      replace29 = replace28.gsub("<Note/>", "<Notes/>")
-      replace30 = replace29.gsub("<Title_Alternative/>", "<Alternate_Title/>")
-      replace31 = replace30.gsub("<Call_Number/>", "<Local_Call_Number/>")
-      replace32 = replace31.gsub("<Audio_Filename/>", "<File_Name/>")
-      replace33 = replace32.gsub("<Video_Filename/>", "<File_Name/>")
-      replace34 = replace33.gsub("<metadata>", "<metadata>\n  <manifest>\n    <contentdm_collection_id>#{collection_file_name}</contentdm_collection_id>\n    <Rails_Root>#{Rails.root}</Rails_Root>\n    <foxml_dir>#{target_dir}</foxml_dir>\n  </manifest>")
+      # relacement array
+      replacements = [
+        #Strip out any bad keying from CDM
+        ["&amp<", "<"],
+        ["&quot<", "<"],
+        ["", ""],
+
+        #Normalize inconsistent CDM metadata vocabulary
+        #So ugly -- remove when vocab is normalized by staff
+        ["<Filename>", "<File_Name>"],
+        ["<Created_by>", "<Created>"],
+        ["<Personal_Name>", "<Personal_Names>"],
+        ["<Organization>", "<Organization_Building>"],
+        ["<Organization-Building>", "<Organization_Building>"],
+        ["<Note>", "<Notes>"],
+        ["<Title_Alternative>", "<Alternate_Title>"],
+        ["<Call_Number>", "<Local_Call_Number>"],
+        ["<Audio_Filename>", "<File_Name>"],
+        ["<Video_Filename>", "<File_Name>"],
+
+        ["</Filename>", "</File_Name>"],
+        ["</Created_by>", "</Created>"],
+        ["</Personal_Name>", "</Personal_Names>"],
+        ["</Organization>", "</Organization_Building>"],
+        ["</Organization-Building>", "</Organization_Building>"],
+        ["</Note>", "</Notes>"],
+        ["</Title_Alternative>", "</Alternate_Title>"],
+        ["</Call_Number>", "</Local_Call_Number>"],
+        ["</Audio_Filename>", "</File_Name>"],
+        ["</Video_Filename>", "</File_Name>"],
+
+        ["<Filename/>", "<File_Name/>"],
+        ["<Created_by/>", "<Created/>"],
+        ["<Personal_Name/>", "<Personal_Names/>"],
+        ["<Organization/>", "<Organization_Building/>"],
+        ["<Organization-Building/>", "<Organization_Building/>"],
+        ["<Note/>", "<Notes/>"],
+        ["<Title_Alternative/>", "<Alternate_Title/>"],
+        ["<Call_Number/>", "<Local_Call_Number/>"],
+        ["<Audio_Filename/>", "<File_Name/>"],
+        ["<Video_Filename/>", "<File_Name/>"],
+        ["<metadata>", "<metadata>\n  <manifest>\n    <contentdm_collection_id>#{collection_file_name}</contentdm_collection_id>\n    <Rails_Root>#{Rails.root}</Rails_Root>\n    <foxml_dir>#{target_dir}</foxml_dir>\n  </manifest>"]
+      ]
+
+      # perform replacement
+      replacements.each do |r|
+        doc1 = doc
+        doc = doc1.gsub(r.first, r.last)
```

Comment:
I don't understand this variable swapping. Is there a reason we can't just do  `doc.gsub!(r.first, r.last)` ?